internal/services/api/post_deployment.go

```diff
@@ -124,6 +126,12 @@ func PostDeploymentHandlerFunc(
 		}
 
 		go func() {
+			// take ownership of this deployment when we start, release when we are done.
+			// release will first check to make sure we still have ownership before releasing
+			// someone else's ownership
+			deployment.ActiveDeploymentRegistry.Set(newState.SaveName, string(newState.LocalID))
```

Comment:
I wasn't getting _any_ file writes on my machine when testing this locally. I did some debugging and `newState.SaveName` was an empty string: `''`

I am getting the `path.String()` correctly in the `WriteFile` method, but is this the correct thing we should be setting here?

Reply:
Fixed now in latest commits.